### PR TITLE
Add `maxQueueSize` to limit the number of unawaited events sent to Sentry

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Limit the max number of unawaited tasks that can be executed by Sentry in a tight loop.

## :bulb: Motivation and Context

The goal of this PR is to evaluate if this approach can be used to solve #408.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- [ ] Evaluate if this approach is feasible.
- [ ] Move to transport implementations?
- [ ] Record discard reason?
- [x] Error handling

### CODE REVIEW

**Comment by denrase** [comment]
@buenaflor This approach should stop the case wehre users capture errors in a tight loop without awaiting the result. The piled up tasks should then be handled by the event loop.

File: dart/lib/src/transport/task_queue.dart [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  int _queueCount = 0;
+
+  Future<T> enqueue(Task<T> task, T fallbackResult) async {
+    if (_queueCount >= _maxQueueSize) {
+      _logger(SentryLevel.warning,
+          'Task dropped due to backpressure. Avoid capturing in a tight loop.');
+      return fallbackResult;
+    } else {
+      _queueCount++;
+      final result = await task();
```
Comment (buenaflor) [inline dart/lib/src/transport/task_queue.dart]: If a task throws an exception, it seems that `_queueCount` would not be decremented

maybe this is safer:
```dart
  _queueCount++;
  try {
    final result = await task();
    return result;
  } finally {
    _queueCount--;
  }
```

**Review by buenaflor — COMMENTED** [review]
left a comment but other than that I think it looks good

**Comment by denrase** [comment]
@stefanosiano Could you also take a 👀 here? 🙇

**Review by stefanosiano — APPROVED** [review]

**Pull request merged**